presto-main/src/main/java/com/facebook/presto/metadata/FunctionManager.java

```diff
@@ -219,13 +227,15 @@ private FunctionHandle resolveFunction(Optional<Session> session, QualifiedName
             return new BuiltInFunctionHandle(getMagicLiteralFunctionSignature(type));
         }
 
-        throw new PrestoException(FUNCTION_NOT_FOUND, constructFunctionNotFoundErrorMessage(name.toString(), parameterTypes, candidates));
+        throw new PrestoException(FUNCTION_NOT_FOUND, constructFunctionNotFoundErrorMessage(functionName.toString(), parameterTypes, candidates));
     }
 
     @Override
     public FunctionMetadata getFunctionMetadata(FunctionHandle functionHandle)
     {
-        return functionNamespaces.get(functionHandle.getFunctionNamespace()).getFunctionMetadata(functionHandle);
+        Optional<FunctionNamespaceManager<?>> functionNamespaceManager = getServingFunctionNamespaceManager(functionHandle.getFunctionNamespace());
+        checkArgument(functionNamespaceManager.isPresent(), "Cannot find function namespace for '%s'", functionHandle.getFunctionNamespace());
```

Comment:
We already have a function handle here, how could there be no FunctionNamespaceManager? Why do we need this new logic?

Reply:
The `functionNamespaces` variables stores the mapping from prefix to manager, and it is acceptable that the prefix only contains the catalog.